controllers/admin/AdminCartRulesController.php

```diff
@@ -530,6 +526,21 @@ public function getProductRuleDisplay($product_rule_group_id, $product_rule_id,
         return $this->createTemplate('product_rule.tpl')->fetch();
     }
 
+    public function populateCategories(&$flatCategories, $currentCategoryTree, $currentPath = '')
```

Comment:
You should type hint it

Reply:
done

Reply:
@PierreRambaud I don't think that a recursive call can be made without reference.

Reply:
Of course it can, everything you can do with reference can be done without it :)

Reply:
done